tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -40,121 +42,147 @@
+            super().__init__(vm=vm, paths=paths, connection=connection)
+            # These are initialized by AgentTestSuite._set_context().
+            # The same runbook execution may execute multiple test suites concurrently. We use 'setup_lock' to ensure
+            # only 1 thread executes setup.
+            self.node: Node = None
+            self.runbook_name: str = None
+            self.suite_name: str = None
+
     def __init__(self, metadata: TestSuiteMetadata) -> None:
         super().__init__(metadata)
-        # The context is initialized by execute()
-        self.__context: AgentLisaTestContext = None
-
-    @property
-    def context(self) -> AgentLisaTestContext:
-        if self.__context is None:
-            raise Exception("The context for the AgentTestSuite has not been initialized")
-        return self.__context
+        # The context is initialized by _set_context() via the call to execute()
+        self.__context: AgentTestSuite.Context = None
 
     def _set_context(self, node: Node):
+        connection_info = node.connection_info
         node_context = get_node_context(node)
-
         runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        # Remove the resource group and node suffix, e.g. "e1-n0" in "lisa-20230110-162242-963-e1-n0"
+        runbook_name = re.sub(r"-\w+-\w+$", "", runbook.name)
 
-        self.__context = AgentLisaTestContext(
-            VmIdentifier(
+        self.__context = AgentTestSuite.Context(
+            vm=VmIdentifier(
                 location=runbook.location,
                 subscription=node.features._platform.subscription_id,
                 resource_group=node_context.resource_group_name,
-                name=node_context.vm_name
-            ),
-            node
-        )
+                name=node_context.vm_name),
+            paths=AgentTestContext.Paths(
+                # The runbook name is unique on each run, so we will use different working directory every time
+                working_directory=Path().home()/"tmp"/runbook_name,
+                remote_working_directory=Path('/home')/connection_info['username']),
+            connection=AgentTestContext.Connection(
+                ip_address=connection_info['address'],
+                username=connection_info['username'],
+                private_key_file=connection_info['private_key_file'],
+                ssh_port=connection_info['port']))
 
-    def _setup(self) -> None:
-        """
-        Prepares the test suite for execution
-        """
-        log.info("Test Node: %s", self.context.vm.name)
-        log.info("Resource Group: %s", self.context.vm.resource_group)
-        log.info("Working directory: %s", self.context.working_directory)
+        self.__context.node = node
+        self.__context.suite_name = f"{self._metadata.full_name}:{runbook.marketplace.offer}-{runbook.marketplace.sku}"
 
-        if self.context.working_directory.exists():
```

Comment:
The cleanup of the working directory was getting in the way of debugging, since sometimes one needs to check its contents. Instead of cleaning up, now I use a unique working directory per runbook execution.

The cleanup was mainly targeted to the developer scenario, so with these changes development machines may start accumulating many test files. That can be handled by our helper scripts.